share/spack/spack-completion.bash

```diff
@@ -80,1266 +104,1448 @@ _bash_completion_spack () {
+        _environments
     fi
 }
 
-_spack_env_view () {
+_spack_env_view() {
     if $list_options
     then
-        compgen -W "-h --help" -- "$cur"
+        SPACK_COMPREPLY="-h --help"
     else
-        compgen -W "regenerate enable disable" -- "$cur"
+        SPACK_COMPREPLY=""
     fi
 }
 
-_spack_extensions () {
+_spack_extensions() {
     if $list_options
     then
-        compgen -W "-h --help -l --long -L --very-long -d --deps -p --paths -s --show -v --view" -- "$cur"
+        SPACK_COMPREPLY="-h --help -l --long -L --very-long -d --deps -p --paths -s --show -v --view"
     else
-        compgen -W "aspell go-bootstrap go icedtea jdk kim-api lua matlab mofem-cephas octave openjdk perl python r ruby rust tcl yorick" -- "$cur"
+        _extensions
     fi
 }
 
-_spack_fetch () {
+_spack_fetch() {
     if $list_options
     then
-        compgen -W "-h --help -n --no-checksum -m --missing -D --dependencies" -- "$cur"
+        SPACK_COMPREPLY="-h --help -n --no-checksum -m --missing -D --dependencies"
     else
-        compgen -W "$(_all_packages)" -- "$cur"
+        _all_packages
     fi
 }
 
-_spack_find () {
+_spack_find() {
     if $list_options
     then
-        compgen -W "-h --help --format --json -d --deps -p --paths --groups --no-groups -l --long -L --very-long -t --tags -c --show-concretized -f --show-flags --show-full-compiler -x --explicit -X --implicit -u --unknown -m --missing -v --variants -M --only-missing --deprecated --only-deprecated -N --namespace --start-date --end-date" -- "$cur"
+        SPACK_COMPREPLY="-h --help --format --json -d --deps -p --paths --groups --no-groups -l --long -L --very-long -t --tags -c --show-concretized -f --show-flags --show-full-compiler -x --explicit -X --implicit -u --unknown -m --missing -v --variants -M --only-missing --deprecated --only-deprecated -N --namespace --start-date --end-date"
     else
-        compgen -W "$(_installed_packages)" -- "$cur"
+        _installed_packages
     fi
 }
 
-_spack_flake8 () {
+_spack_flake8() {
     if $list_options
     then
-        compgen -W "-h --help -b --base -k --keep-temp -a --all -o --output -r --root-relative -U --no-untracked" -- "$cur"
+        SPACK_COMPREPLY="-h --help -b --base -k --keep-temp -a --all -o --output -r --root-relative -U --no-untracked"
+    else
+        SPACK_COMPREPLY=""
     fi
 }
 
-_spack_gpg () {
+_spack_gc() {
+    SPACK_COMPREPLY="-h --help -y --yes-to-all"
+}
+
+_spack_gpg() {
     if $list_options
     then
-        compgen -W "-h --help" -- "$cur"
+        SPACK_COMPREPLY="-h --help"
     else
-        compgen -W "verify trust untrust sign create list init export" -- "$cur"
+        SPACK_COMPREPLY="verify trust untrust sign create list init export"
     fi
 }
 
-_spack_gpg_verify () {
+_spack_gpg_verify() {
     if $list_options
     then
-        compgen -W "-h --help" -- "$cur"
+        SPACK_COMPREPLY="-h --help"
     else
-        compgen -W "$(installed_packages)" -- "$cur"
```

Comment:
This was a bug, we were missing the leading underscore. Same below for `_spack_gpg_sign`.